File: ChainJava/src/chain/MainClass.java
package chain;

import chain.employees.ClubPresident;
import chain.employees.Coach;
import chain.employees.Employee;
import chain.employees.SportingDirector;
import chain.requests.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created by Karolina_2 on 2015-04-01.
 */
public class MainClass {

     public static void main(String [] args) {
        List<Request> listOfRequests = gatherRequests();

        Employee clubPresident = new ClubPresident();
        Employee sportingDirector = new SportingDirector(clubPresident);
        Employee coach = new Coach(sportingDirector);
        for(Request request : listOfRequests){
            coach.receiveARequest(request);
        }
    }

    private static List<Request> gatherRequests(){
        List<Request> requests = new ArrayList<Request>();
        requests.add(new ContractRequest("x", new Date(), new Date()));
        requests.add(new DressingRoomRepairRequest("y", new Date()));
        requests.add(new TransferRequest("z", new Date()));
        requests.add(new PersonalTrainingRequest("a", new Date()));
        requests.add(new TransferRequest("b", new Date()));
        return requests;
    }
}

File: ChainC#/Chain/MainClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZTP_Zadanie3.chain.wnioski;
using ZTP_Zadanie3.pracownicy;

namespace ZTP_Zadanie3
{
    class MainClass
    {
        public static void Main(string[] args)
        {
            List<Request> listOfRequests = gatherRequests();

            Employee clubPresident = new ClubPresident();
            Employee sportingDirector = new SportingDirector(clubPresident);
            Employee coach = new Coach(sportingDirector);
            foreach(Request request in listOfRequests){
                coach.receiveARequest(request);
            }
            Console.ReadKey();
        }

        private static List<Request> gatherRequests()
        {
            List<Request> requests = new List<Request>();
            requests.Add(new ContractRequest("x", new DateTime(), new DateTime()));
            requests.Add(new DressingRoomRepairRequest("y", new DateTime()));
            requests.Add(new TransferRequest("z", new DateTime()));
            requests.Add(new PersonalTrainingRequest("a", new DateTime()));
            requests.Add(new TransferRequest("b", new DateTime()));

            return requests;
        }
    }
}
